sound/soc/sof/loader.c

```diff
@@ -501,11 +595,30 @@ int snd_sof_load_firmware_raw(struct snd_sof_dev *sdev)
 			fw_filename);
 	}
 
-	/*
-	 * by default firmware should be loaded in a whole,
-	 * so don't skip the beginning during loading FW to DSP memory.
-	 */
-	plat_data->fw_offset = 0;
+	/* check for extended manifest */
+	ext_man_size = snd_sof_ext_man_size(plat_data->fw);
+	if (ext_man_size > 0) {
+		ret = snd_sof_fw_ext_man_parse(sdev, plat_data->fw);
+
+		/* when no error occurred, drop extended manifest */
+		if (!ret)
+			plat_data->fw_offset = ext_man_size;
+		else
+			dev_err(sdev->dev, "error: firmware %s contains unsupported or invalid extended manifest: %d\n",
+				fw_filename, ret);
+	} else if (!ext_man_size) {
+		/*
+		 * there is no any extended manifest,
+		 * firmware should be loaded in a whole,
+		 * so don't skip the beginning during loading FW to DSP memory.
+		 */
```

Comment:
Can simply say "No extended manifest, so nothing to skip during FW load"

Reply:
ok